### PR TITLE
Adding JMX Monitoring for Human Task

### PR DESCRIPTION
**Pull request by shivabalachandran**

### CODE REVIEW

File: components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/mgt/HTQueryBuildHelper.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*
+ *
+ *   Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (cnapagoda) [inline components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/mgt/HTQueryBuildHelper.java]: Year should be 2015?

File: components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/mgt/HTQueryBuildHelperImpl.java [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+                            humanTaskServer.getServerConfig();
+                            HTQueryBuilder htQueryBuilder = new HTQueryBuilder();
+                            EntityManager em = humanTaskDAOConnectionFactory.getConnection().getEntityManager();
+                            Query query = htQueryBuilder.buildQueryToFindTaskInstances(taskname, em);
+                            tasklist[0] = query.getResultList();
+                            taskCountTemp[0] = tasklist[0].size();
+                            return null;
+                        }
+                    });
+        } catch (Exception ex) {
```
Comment (cnapagoda) [inline components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/mgt/HTQueryBuildHelperImpl.java]: Is there any reason to catch top level exception here?

File: components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/mgt/HTQueryBuildHelper.java (hunk 1 above)
Comment (shivabalachandran) [inline components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/mgt/HTQueryBuildHelper.java]: Fixed in the latest commit.

File: components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/mgt/HTQueryBuildHelperImpl.java (hunk 2 above)
Comment (shivabalachandran) [inline components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/mgt/HTQueryBuildHelperImpl.java]: Fixed in the latest commit. Moved all exception catches to the top most level of the code.

**Comment by shivabalachandran** [comment]
Closed PR due to conflicts, created new PR. Check https://github.com/wso2/carbon-business-process/pull/64

**Pull request closed without merging**